crates/evm/evm/src/executors/invariant/shrink.rs

```diff
@@ -0,0 +1,69 @@
+use proptest::bits::{BitSetLike, VarBitSet};
+
+#[derive(Clone, Copy, Debug)]
+struct Shrink {
+    call_index: usize,
+}
+
+/// Shrinker for a call sequence failure.
+/// Iterates sequence call sequence top down and removes calls one by one.
+/// If the failure is still reproducible with removed call then moves to the next one.
+/// If the failure is not reproducible then restore removed call and moves to next one.
+#[derive(Debug)]
+pub(crate) struct CallSequenceShrinker {
+    /// Call sequence to be shrinked.
+    call_sequence: Vec<usize>,
+    /// Call ids contained in current shrinked sequence.
+    included_calls: VarBitSet,
+    /// Current shrinked call id.
+    shrink: Shrink,
+    /// Previous shrinked call id.
+    prev_shrink: Option<Shrink>,
+}
+
+impl CallSequenceShrinker {
+    pub(crate) fn new(call_sequence: Vec<usize>) -> Self {
+        let included_calls = VarBitSet::saturated(call_sequence.len());
+        Self { call_sequence, included_calls, shrink: Shrink { call_index: 0 }, prev_shrink: None }
+    }
+
+    /// Return candidate shrink sequence to be tested, by removing ids from original sequence.
+    pub(crate) fn current(&self) -> Vec<usize> {
+        self.call_sequence
+            .iter()
+            .enumerate()
```

Comment:
is this needed as a vec? Can we not just store the length and use 0..x here?

Reply:
nice! changed it to store call seq length in https://github.com/foundry-rs/foundry/pull/7756/commits/55a44ba8ebb57adb65086789cbf71d3f1f79cd1d